src/pspm_pfm.m

```diff
@@ -0,0 +1,588 @@
+
+% Checking marker channels 
+if strcmpi(model.timeunits,'markers')
+    if ~isfield(model,'marker_chan')
+        model.marker_chan = 'marker';
+    elseif numel(model.marker_chan)==1 && ~isnumeric(model.marker_chan)
+        warning('ID:invalid_input','Marker channels have to be numeric.'); return;
+    elseif numel(model.marker_chan)>1 && ...
+           ( ~iscell(model.marker_chan) || ... 
+             numel(model.marker_chan)~=numel(model.marker_chan) )
+        warning('ID:invalid_input',['Marker channels have to be a cell array', ...
+                                    ' of the same size than ''model.modelfile''.'])
+    end
+end
+
+% Checking standard experimental condition
+std_cond_war_msg = ['The standard experimetal condition must be',...
+                ' either a valid experimental condition or an',...
+                ' index corresponding to it.'];
+if ~isfield(model,'std_exp_cond')
+    model.std_exp_cond = 'none';
+elseif ~ischar(model.std_exp_cond) && ~isnumeric(model.std_exp_cond)
+    warning('ID:invalid_input',std_cond_war_msg); return;
+elseif ischar(model.std_exp_cond)
+    tmp_ind = cellfun(@(x) strcmpi(model.std_exp_cond,x),model.timing{1}.names);
+    if ~any(tmp_ind)
+        warning('ID:invalid_input',std_cond_war_msg); return;
+    end
+    std_exp_cond.name = model.std_exp_cond;
+    std_exp_cond.ind = find(tmp_ind);
+elseif isnumeric(model.std_exp_cond)
+    if model.std_exp_cond < 1 || ...
+       model.std_exp_cond > numel(model.timing{1}.names)
+        warning('ID:invalid_input',std_cond_war_msg); return;
+    end
+    std_exp_cond.name = model.timing{1}.names(model.std_exp_cond);
+    std_exp_cond.ind = model.std_exp_cond;
+end
+clear std_cond_war_msg tmp_ind
+
+%Checking norm
+if ~isfield(model, 'norm')
+    model.norm = 0;
+elseif ~ismember(model.zscore, [0, 1])
+    warning('ID:invalid_input', '''model.zscore'' has to be 0 or 1.'); return;
+end
+
+%%% Checking options %%%
+if ~isfield(options, 'overwrite')
+    options.overwrite = 0;
+elseif ~ismember(options.overwrite, [0, 1])
+    options.overwrite = 0;
+end
+
+if exist(model.modelfile, 'file') && ~(isfield(options, 'overwrite') && options.overwrite == 1)
+    overwrite=menu(sprintf('Model file (%s) already exists. Overwrite?', model.modelfile), 'yes', 'no');
+    if overwrite == 2, return, end
+    options.overwrite = 1;
+end
+
+%%
+%%%%%%%% Loading files %%%%%%%%
+
+fprintf('Computing Pupil Model: %s \n', model.modelfile);
+
+nExpCond = numel(model.timing{1}.names);     % number of experimental conditions
+nFile = numel(model.datafile);      % number of files
+
+% Loading data and sr
+fprintf('Getting data .');
+for iFile = 1:nFile
+    if iscell(model.channel)
+        [sts, ~, data] = pspm_load_data(model.datafile{iFile}, model.channel{iFile});
+    else 
+        [sts, ~, data] = pspm_load_data(model.datafile{iFile}, model.channel);
+    end
+    if sts < 1, warning('ID:load_data_fail', 'Problem encountered while loading data.'); return; end
```

Comment:
I like the idea to use separate error IDs for different errors, which is not done in most parts of PsPM currently. I will open a low priority issue so that we can keep it in mind and slowly proceed in this direction.

Reply:
Great !